components/VM_Arm/src/main.c

```diff
@@ -1270,8 +1271,26 @@ static int main_continued(void)
         }
     }
 
+    int pcpu_assignments = 0;
```

Comment:
Let's avoid assigning 0 and then overwriting this, it gives compiler warnings at the picky level.
```
    unsigned int pcpu_assignments = get_instance_size_pcpus_list();
    if (pcpu_assignments > NUM_VCPUS) {
        pcpu_assignments = NUM_VCPUS;
    }
```

Reply:
Or well, we could even merge this nicely into the CPU node creation at line 1260:

``` 
    /* Create CPUs and DTB node */
    unsigned int pcpu_assignments = get_instance_size_pcpus_list();
    for (int i = 0; i < NUM_VCPUS; i++) {
        vm_vcpu_t *new_vcpu = create_vmm_plat_vcpu(&vm, VM_PRIO - 1);
        assert(new_vcpu);
        if (i < pcpu_list_size) {
            /* Assign VCPU to explicit physical CPU */
            new_vcpu->target_cpu = pcpus[i];
        }
    }

    // ... and not generatethe device tr ee VCPU nodes
```